src/main/webapp/WEB-INF/user.jsp

```diff
@@ -41,11 +42,23 @@
     </form>
     <hr/>
 
+    <% if(isUserLoggedIn){ %>
+    <form id="message-form" action="/messages?recipient=<%= request.getAttribute("user") %>" method="POST" class>
+    <% } else { %>
+    <form id="message-form" action="/messages?recipient=" method="POST" class="hidden">
+    <% } %>
+      Enter a new message:
+      <br/>
+      <textarea name="text" placeholder="Enter a message" id="message-input"></textarea>
+      <br/>
+      <input type="submit" value="Submit">
+    </form>
+
     <div id="message-container">
-    <%  if (true) { %>
+    <%  if (messages.size() == 0) { %>
```

Comment:
Consider using messages.isEmpty() instead. It is a little more readable and more closely matches what you would likely say if you were describing what this code was doing.